compiler/deSugar/PmExpr.hs

```diff
@@ -435,7 +435,7 @@ pprPmExprCon (RealDataCon con) args
     pretty_list :: PmPprM SDoc
     pretty_list = case isNilPmExpr (last list) of
       True  -> brackets . fsep . punctuate comma <$> mapM pprPmExpr (init list)
-      False -> parens   . hcat . punctuate colon <$> mapM pprPmExpr list
+      False -> parens   . hcat . punctuate dcolon <$> mapM pprPmExpr list
```

Comment:
For context, our fork of GHC checks the `DamlSyntax` language extension to pick a colon convention, see [`performNewColonConvention`](https://github.com/digital-asset/ghc/blob/7be2c87f191abb4a13d77171968dcdbe9a25b40e/compiler/main/DynFlags.hs#L1645-L1647). With that in mind, I would suggest adding `list_cons :: SDoc` mirroring [`of_type`](https://github.com/digital-asset/ghc/blob/7be2c87f191abb4a13d77171968dcdbe9a25b40e/compiler/utils/Outputable.hs#L643-L647), allowing us to rewrite this as

```suggestion
      False -> parens   . hcat . punctuate list_cons <$> mapM pprPmExpr list
```

Reply:
to be clear, this doesn't make much of a difference since the `DamlSyntax` extension is always enabled in `damlc`, but this would be more consistent with existing uses of `performNewColonConvention`

Reply:
Oh of course! I saw of_type before and it completely slipped my mind. Will do :)

Reply:
thanks!